ddtrace/internal/datadog/profiling/src/exporter.cpp

```diff
@@ -176,12 +176,13 @@ Uploader::set_runtime_id(std::string_view id)
 }
 
 bool
-Uploader::upload(const Profile* profile)
+Uploader::upload(Profile* profile)
 {
     ddog_prof_Profile_SerializeResult result = ddog_prof_Profile_serialize(profile->ddog_profile, nullptr, nullptr);
+    profile->reset();
```

Comment:
I don't expect an upload function to reset the profile.

Reply:
I did not check if there was a better place to reset the profile.

Reply:
Do we expect to be able to flush the data to multiple sinks? E.g. a file _and_ the network? If not I'd be OK with a reset here.

Reply:
I need to reason about this a bit more.  There's something I don't understand, since I have a reset in the higher-level interface already (but it's in the wrapper for `upload()` called by the client).

@P403n1x87: to your question, I do plan on supporting serialize-to-file just like the current profiler does, and I do anticipate that setting being additive with pushing through the network.

The general comment came up in an earlier review.  I think the main concern is that the semantic meaning of the word "upload" encompasses only a step of what we're doing here.  Will address with a refactor.

Reply:
Some reasons behind my comment: 
- The reset is done just before we start writing to the profile (to ensure the duration of the profile is accurate). 
- This runs in an exporter thread where we would not expect the profile to be updated.

As mentioned by @sanchda there seems to be a reset in the right place, so we will look at this more thoroughly.